laravel-app/app/Http/Controllers/PostController.php

```diff
@@ -88,14 +88,22 @@ public function update(UserPostRequest $request, UserPost $post): RedirectRespon
     {
         $validatedData = $request->validated();
         $updated = $this->postService->updatePost($post, $validatedData);
+        $image = $request->file('image');
 
-        if ($updated) {
-            $success = ['success' => 'Post updated successfully'];
+        if ($image && !$updated) {
+            return redirect()->back()->with('error', 'Failed to update post and/or image');
+        }
 
-            return redirect()->route('post.show', ['post' => $post])->with($success);
+        if ($image instanceof UploadedFile) {
+            $updatedImage = $this->postService->isPostImageUpdatable($post, $image);
+            $successMessage = $updatedImage ? 'Post and image updated successfully' : 'Image updated successfully';
+        } elseif ($updated) {
+            $successMessage = 'Post updated successfully';
+        } else {
+            return redirect()->back()->with('error', 'Failed to update post and/or image');
         }
```

Comment:
You can prevent using else in the if statement. Try to research more

Reply:
Yes sir 🙇